src/build-data/cc/clangcl.txt

```diff
@@ -0,0 +1,102 @@
+macro_name CLANG
```

Comment:
Probably we should make this `CLANGCL` to be able to distinguish vanilla clang from clang-cl. Much like that was [done for Xcode](https://github.com/randombit/botan/pull/4010) a few months ago.

Note that this change is incorporated in the patch I proposed on top.